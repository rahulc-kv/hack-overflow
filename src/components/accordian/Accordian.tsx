import React, { FC } from 'react';

import { ChevronDown } from '@assets/icons';
import { AccordianProps } from './types';

const Accordian: FC<AccordianProps> = props => {
  const {
    styleConfig,
    contentConfig,
    conditionalFlag,
    dataTestId = 'accordianHeading'
  } = props;

  const {
    accordianStyle = '',
    accordianHeadingWrapperStyle = '',
    accordianIconStyle
  } = styleConfig;
  const { accordianBody, accordianHeading, handleOnClickHeading } =
    contentConfig;
  const {
    chevronDownPresent = true,
    showDetails,
    disableOnClick = false
  } = conditionalFlag;

  const handleOnClick = () => {
    if (disableOnClick) return;
    handleOnClickHeading();
  };

  return (
    <div className={`${accordianStyle} flex flex-col p-5 w-full rounded-md`}>
      <div
        className={`${accordianHeadingWrapperStyle} flex justify-between items-center w-full cursor-pointer`}
        onClick={handleOnClick}
        data-testid={dataTestId}>
        <div className="flex w-full">{accordianHeading}</div>
        {chevronDownPresent && (
          <ChevronDown
            className={`${showDetails ? 'transform  rotate-180 ' : ''}
             ${
               accordianIconStyle ? accordianIconStyle : 'ml-2'
             } opacity-100 duration-1000`}
          />
        )}
      </div>
      <div
        className={`overflow-hidden  duration-1000 transition-height ease opacity-100
       ${showDetails ? 'visible max-h-[2000px]' : ' max-h-0 invisible'}`}>
        {accordianBody}
      </div>
    </div>
  );
};

export default Accordian;
