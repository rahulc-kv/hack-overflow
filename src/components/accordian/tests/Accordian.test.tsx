import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Accordian from '../Accordian';

const handleClick = jest.fn();

test('Accordian Rendered', () => {
  const component = renderer
    .create(
      <Accordian
        dataTestId="accordianHeading"
        styleConfig={{
          accordianStyle: '',
          accordianIconStyle: ''
        }}
        contentConfig={{
          accordianHeading: '',
          accordianBody: '',
          handleOnClickHeading: handleClick
        }}
        conditionalFlag={{
          showDetails: true,
          chevronDownPresent: true
        }}
      />
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});

test('Accordian Clicked', () => {
  render(
    <Accordian
      dataTestId="accordianHeading"
      styleConfig={{
        accordianStyle: '',
        accordianIconStyle: ''
      }}
      contentConfig={{
        accordianHeading: '',
        accordianBody: '',
        handleOnClickHeading: handleClick
      }}
      conditionalFlag={{
        showDetails: true,
        chevronDownPresent: true
      }}
    />
  );
  fireEvent.click(screen.getByTestId('accordianHeading'));
  expect(handleClick).toHaveBeenCalledTimes(1);
});
