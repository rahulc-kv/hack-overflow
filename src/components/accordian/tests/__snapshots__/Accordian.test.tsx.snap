// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Rendered 1`] = `
<div
  className=" flex flex-col p-5 w-full rounded-md"
>
  <div
    className=" flex justify-between items-center w-full cursor-pointer"
    data-testid="accordianHeading"
    onClick={[Function]}
  >
    <div
      className="flex w-full"
    >
      
    </div>
    <svg
      className="transform  rotate-180 
             ml-2 opacity-100 duration-1000"
    >
      chevronDown.svg
    </svg>
  </div>
  <div
    className="overflow-hidden  duration-1000 transition-height ease opacity-100
       visible max-h-[2000px]"
  >
    
  </div>
</div>
`;
